etc/travis/DeploymentTester.py

```diff
@@ -0,0 +1,124 @@
+import pycurl
+import sys
+import time
+from pycurl import Curl
+
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.subscription import SubscriptionClient
+from msrestazure.azure_active_directory import ServicePrincipalCredentials
+
+from travis.Configuration import Configuration
+
+
+class DeploymentTester:
+    @staticmethod
+    def elapsed(since):
+        elapsed = int(time.time() - since)
+        elapsed = '{:02d}:{:02d}:{:02d}'.format(elapsed // 3600, (elapsed % 3600 // 60), elapsed % 60)
+        return elapsed
+
+    def __init__(self):
+        self.config = Configuration()
+        self.deployment = None
+
+        self.credentials = None
+        """:type : ServicePrincipalCredentials"""
+
+        self.resource_client = None
+        """:type : ResourceManagementClient"""
+
+    def run(self):
+        self.check_configuration()
+        self.login()
+        self.create_resource_group()
+        self.validate()
+        self.deploy()
+        self.moodle_smoke_test()
+        print('\n\nJob done!')
+
+    def check_configuration(self):
+        print('\nChecking configuration...')
+        if not self.config.is_valid():
+            print('No Azure deployment info given, skipping test deployment and exiting.')
+            print('Further information: https://github.com/Azure/Moodle#automated-testing-travis-ci')
+            sys.exit()
+        artifacts_location = self.config.deployment_properties['parameters']['_artifactsLocation']
+        print('- Detected "_artifactsLocation": ' + artifacts_location['value'])
+        print("(all check)")
+
+    def login(self):
+        print('\nLogging in...')
+        self.credentials = ServicePrincipalCredentials(
+            client_id=self.config.client_id,
+            secret=self.config.secret,
+            tenant=self.config.tenant_id,
+        )
+        print('(got credentials)')
+        subscription_client = SubscriptionClient(self.credentials)
+        subscription = next(subscription_client.subscriptions.list())
+        print('(found subscription)')
```

Comment:
@hosungsmsft These lines should do the trick to fetch the first subscription (most likely the only one) available for a given service principal so I removed the configurable option.

Reply:
Excellent! Thanks much for your research and improvement!